### PR TITLE
v12 epn subtyping (3/N)

### PR DESCRIPTION
**Pull request by ewafula**
#### What scientific question is your analysis addressing?
This PR runs EPN subtyping for v12

#### What was your approach?
Made the following changes:
- Updated script 01 to that assigns disease groups to include high-confidence methylation EPN subtypes from the histologies base file 
- updated gene symbols utilized within the module to Gencode v39
```
NEW	OLD
EZHIP	CXorf67
H3-3A	H3F3A
H3-3B	H3F3B
H3C2	HIST1H3B
H3C3	HIST1H3C
H3C14	HIST2H3C
ZFTA	C11orf95
```
- excluded  the following gene fusions that are no longer required
```
LTBP3--RELA 
ZTFA--YAP1
```
- Kept subtyping as was previously performed (i.e., gene expression and scoring gene fusions, focal cnv, etc. ) but assigned EPN subgroups from methylation classification where subgroups have not been determined (i.e., `EPN, To be classified`).
- README updated.

#### What GitHub issue does your pull request address?
Part of the v12 release

#### Which areas should receive a particularly close look?
Please review scripts 01, 02, and 03 in detail to check if they capture everything accurately and logic is correct.

#### Is there anything that you want to discuss further?
NA

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes

#### What types of results are included (e.g., table, figure)?
Updates EPN subgroups table output.

#### What is your summary of the results?

612 EPN bs ids (DNA = =212, RNA == 193, and Methly = = 207)
230 patients
398 tumors subtyped:

```r
 subgroup                  n
   <chr>                 <int>
 1 EPN, PF A               267
 2 EPN, PF B                 9
 3 EPN, PF SE                1
 4 EPN, SP                  29
 5 EPN, SP-MYCN              2
 6 EPN, SP-SE B              1
 7 EPN, ST YAP1              8
 8 EPN, ST ZFTA             81
 9 EPN, To be classified    61
10 N/A                      10
> 
```

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/molecular-subtyping-EPN/01-assign-disease-group.R [hunk 1]
```diff
@@ -30,35 +30,56 @@ EP = pbta_histologies %>%
...
   dplyr::select(Kids_First_Biospecimen_ID, Kids_First_Participant_ID, sample_id, primary_site, CNS_region) %>%
   dplyr::rename("Kids_First_Biospecimen_ID_RNA" = "Kids_First_Biospecimen_ID")
 
 # filter for DNA samples 
 WGS_dnaseqsamples = EP %>%
-  filter(experimental_strategy == "WGS") %>%
-  dplyr::select(Kids_First_Biospecimen_ID, Kids_First_Participant_ID, sample_id, primary_site, CNS_region) %>%
+  filter(experimental_strategy == "WGS" | experimental_strategy == "WXS" | experimental_strategy == "Targeted Sequencing",
+         is.na(RNA_library)) %>%
+  dplyr::select(Kids_First_Biospecimen_ID, Kids_First_Participant_ID, sample_id, primary_site, CNS_region, ) %>%
```
Comment (zzgeng) [inline analyses/molecular-subtyping-EPN/01-assign-disease-group.R]:
```suggestion
  dplyr::select(Kids_First_Biospecimen_ID, Kids_First_Participant_ID, sample_id, primary_site, CNS_region) %>%
```
Everything works fine but found a typo here.

File: analyses/molecular-subtyping-EPN/03-summary.Rmd [hunk 2]
```diff
@@ -191,7 +192,7 @@ viewDataTable(EPN_final %>%
 
 <!-- 1. This next part assigns RELA and YAP1 subgroups based on a set of filters.   -->
```
Comment (rjcorb) [inline analyses/molecular-subtyping-EPN/03-summary.Rmd]: Lines 191-247 are obsolete and can be deleted

File: analyses/molecular-subtyping-EPN/README.md [hunk 3]
```diff
@@ -38,14 +38,16 @@ This above  script is designed to change to this directory to run, so it should
 
     A new column named `subgroup` is added to the input table and saved in `results/EPN_all_data_withsubgroup.tsv`.
 
-    This script prioritizes features of subgroups first and does not assign those samples to any other subgroups. For example, samples where `disease_group == 'spinal'` are prioritized for the `EPN, SP` or `EPN, SP-MYCN` subgroup, and are not assigned to any other groups. Samples are tested for assignment to subgroups in the following prioritized order:
+    This script prioritizes features of subgroups first and does not assign those samples to any other subgroups. For example, samples where `disease_group == 'spinal'` are prioritized for the `EPN, SP` or `EPN, SP-MYCN` subgroup, and are not assigned to any other groups. Samples are tested for assignment to subgroups in the following prioritized order shown below. Subgroups inferred from methylation classification are only included where subgroups are determined (i.e., EPN, To be classified).
```
Comment (rjcorb) [inline analyses/molecular-subtyping-EPN/README.md]:
```suggestion
    This script prioritizes features of subgroups first and does not assign those samples to any other subgroups. For example, samples where `disease_group == 'spinal'` are prioritized for the `EPN, SP` or `EPN, SP-MYCN` subgroup, and are not assigned to any other groups. Samples are tested for assignment to subgroups in the following prioritized order shown below. Subgroups inferred from methylation classification are only included where subgroups are not determined (i.e., EPN, To be classified).
```

**Review by rjcorb — CHANGES_REQUESTED** [review]
The code for spinal subtyping in lines 74-76 of `03-summary.Rmd`needs to be changed to:

```
EPN_final <- EPN_final %>%
  mutate(subgroup = ifelse(disease_group == "spinal" & grepl("amplification", consensus_focal_CN_MYCN), "EPN, SP-MYCN", 
                           ifelse(disease_group == "spinal", "EPN, SP", subgroup)))
```

the current `ifelse` statement that specifies `consensus_focal_CN_MYCN == "amplification"` is the cause of `NAs` in the final subgroup column. This was probably an oversight on my part when I added the spinal classification a few months ago! Otherwise, everything looks good to me.

File: analyses/molecular-subtyping-EPN/03-summary.Rmd (hunk 2 above)
Comment (ewafula) [reply analyses/molecular-subtyping-EPN/03-summary.Rmd]: deleted

**Comment by ewafula** [comment]
@rjcorb and @zzgeng , I  have made the review changes you recommended. Ready for you to check again.

**Review by zzgeng — APPROVED** [review]
Everything works fine and logic is solid. Approving!

**Review by rjcorb — APPROVED** [review]

**Comment by jharenza** [comment]
@ewafula and @rjcorb : looking at these results, I think we should update the logic for spinal ependymoma:

If spinal and MYCN amp --> `EPN, SP-MYCN`
If spinal and has methylation classification of SP or MPE, choose the methyl subtype, else --> To be classified.

Saying this because as I am reviewing WHO 2021, the SP subtype is grade 3, while MPE (myxopapillary, also can be in spine) is grade I and I don't want us to mis-classify those up to SP. 

Can you make that change?

File: analyses/molecular-subtyping-EPN/01-assign-disease-group.R [hunk 4]
```diff
@@ -30,35 +30,56 @@ EP = pbta_histologies %>%
...
+                (grepl("EPN_", dkfz_v12_methylation_subclass) & dkfz_v12_methylation_subclass_score >= 0.8)) %>%
+  dplyr::mutate(molecular_subtype_methyl = case_when(grepl("EPN_PFA", dkfz_v12_methylation_subclass) ~ "EPN, PF A",
+                                                     grepl("EPN_PFB", dkfz_v12_methylation_subclass) ~ "EPN, PF B",
+                                                     grepl("EPN_ST_ZFTA", dkfz_v12_methylation_subclass) ~ "EPN, ST ZFTA",
+                                                     dkfz_v12_methylation_subclass == "EPN_MPE" ~ "EPN, MPE",
+                                                     dkfz_v12_methylation_subclass == "EPN_PF_SE" ~ "EPN, PF SE",
+                                                     dkfz_v12_methylation_subclass == "EPN_SPINE" ~ "EPN, SP",
+                                                     dkfz_v12_methylation_subclass == "EPN_SPINE_MYCN" ~ "EPN, SP-MYCN",
+                                                     dkfz_v12_methylation_subclass == "EPN_SPINE_SE_A" ~ "EPN, SP-SE A",
+                                                     dkfz_v12_methylation_subclass == "EPN_SPINE_SE_B" ~ "EPN, SP-SE B",
```
Comment (jharenza) [inline analyses/molecular-subtyping-EPN/01-assign-disease-group.R]:
```suggestion
                                                     dkfz_v12_methylation_subclass == "EPN_SPINE_SE_A" ~ "EPN, SP-SE",
                                                     dkfz_v12_methylation_subclass == "EPN_SPINE_SE_B" ~ "EPN, SP-SE",
```

minor update request here

File: analyses/molecular-subtyping-EPN/README.md [hunk 5]
```diff
@@ -38,14 +38,16 @@ This above  script is designed to change to this directory to run, so it should
...
+    This script prioritizes features of subgroups first and does not assign those samples to any other subgroups. For example, samples where `disease_group == 'spinal'` are prioritized for the `EPN, SP` or `EPN, SP-MYCN` subgroup, and are not assigned to any other groups. Samples are tested for assignment to subgroups in the following prioritized order shown below. Subgroups inferred from methylation classification are only included where subgroups are not determined (i.e., EPN, To be classified).
     1) EPN, SP-MYCN
     2) EPN, SP
     3) EPN, ST ZFTA
     4) EPN, ST YAP1
     5) EPN, PF A 
     6) EPN, PF B
-    7) EPN, To be classified
+    7) EPN, PF SE (methylation classificaton)
+    8) EPN, SP-SE B (methylation classificaton)
```
Comment (jharenza) [inline analyses/molecular-subtyping-EPN/README.md]:
```suggestion
    8) EPN, SP-SE (methylation classificaton)
```

**Pull request merged**